src/genericitems/list.h

```diff
@@ -28,30 +29,163 @@ class GenericList
+
+        // Find item in the list
+        auto *item = find(varName);
+        if (!item)
+        {
+            Messenger::printVerbose("No item named '{}' in list - default value item will be returned.\n", varName);
+            static T dummy;
+            dummy = defaultValue;
+            if (found != nullptr)
+                (*found) = false;
+            return dummy;
+        }
+
+        // Cast to correct type
+        auto *castItem = dynamic_cast<const GenericItemContainer<T> *>(item);
+        if (!castItem)
+            throw std::runtime_error(
+                fmt::format("GenericListHelper::value({}) failed, because the target item is of the wrong type.", name));
+
+        if (found != nullptr)
+            (*found) = true;
+        return castItem->data();
+    }
+    // Retrieve named item from specified list as template-guided type, assuming that it is going to be modified
+    template <class T>
+    T &retrieve(std::string_view name, std::string_view prefix = "", T defaultValue = T(), bool *found = nullptr)
+    {
+        // Construct full name
+        std::string varName = prefix.empty() ? std::string(name) : fmt::format("{}_{}", prefix, name);
+
+        // Find item in the list
+        auto *item = find(varName);
+        if (!item)
+        {
+            Messenger::printVerbose("No item named '{}' in list - default value item will be returned.\n", varName);
+            static T dummy;
+            dummy = defaultValue;
+            if (found != nullptr)
+                (*found) = false;
+            return dummy;
+        }
+
+        // Cast to correct type
+        auto *castItem = dynamic_cast<GenericItemContainer<T> *>(item);
+        if (!castItem)
+            throw std::runtime_error(
+                fmt::format("GenericListHelper::retrieve({}) failed, because the item is not of the specified type.", name));
+
+        // Bump the version of the item
+        item->bumpVersion();
+
+        if (found != nullptr)
+            (*found) = true;
+        return castItem->data();
+    }
+    // Create or retrieve named item from specified list as template-guided type
+    template <class T> T &realise(std::string_view name, std::string_view prefix = "", int flags = -1, bool *created = nullptr)
+    {
+        // Construct full name
+        std::string varName = prefix.empty() ? std::string(name) : fmt::format("{}_{}", prefix, name);
+
+        // Find item in the list - if it isn't there, create it and return
+        auto *item = find(varName);
+        if (!item)
+        {
+            if (created != nullptr)
+                (*created) = true;
+            return add<T>(name, prefix, flags);
+        }
+
+        // Cast to correct type
+        auto *castItem = dynamic_cast<GenericItemContainer<T> *>(item);
+        if (!castItem)
+            throw std::runtime_error(fmt::format(
+                "GenericListHelper::realise({}) failed, because the item couldn't be cast to the desired type.", name));
```

Comment:
Very minor, but do these throws need to be updated given the refactor/moving about of the code. For instance, something like
```suggestion
                "GenericList.realise({}) failed, because the item couldn't be cast to the desired type.", name));
```